jme3-lwjgl3/src/main/java/com/jme3/system/lwjgl/LwjglWindow.java

```diff
@@ -534,6 +540,18 @@ protected void runLoop() {
             throw new IllegalStateException();
         }
 
+        // Update the frame buffer size from 2nd frame since the initial value
+        // of frame buffer size from glfw maybe incorrect when HiDPI display is in use
+        if (framesAfterContextStarted < 2) {
+            framesAfterContextStarted++;
+            if (framesAfterContextStarted == 2) {
+                int[] width = new int[1];
+                int[] height = new int[1];
+                glfwGetFramebufferSize(window, width, height);
+                listener.reshape(width[0], height[0]);
```

Comment:
Well, nobody likes this hack. But you did a very good job testing it and reporting it to upstream. Maybe this, if there is no other choice... should be conditional? Reshape might be heavy and there is no reason to call it every time on frame 2.

For these HiDPI displays I know that the initial size is wrong, so this is a fix for https://github.com/jMonkeyEngine/jmonkeyengine/issues/893?

Reply:
So could it be somehow changed that when everything is done do a glfwPollEvents()? But what I'm really after is, if the currentDimension <> newDimensions -> reshape? Not that reshape every application on frame 2.

Reply:
@jmecn Do you understand what tonihele is requesting?

Reply:
@stephengold @tonihele 
I add a resolution check before that reshape, would it be better?